code-challenge/Models/Discover/Discover.swift

```diff
@@ -0,0 +1,53 @@
+import Foundation
+
+struct Discover: Decodable {
+    let identifier: Int
+    let title: String
+    let votes: Int
+    let isVideo: Bool
+    let avarageVote: Double
+    let popularity: Double
+    let poster: String
+    let originalLanguage: String
+    let originalTitle: String
+    let genre: [Int]
+    let backdrop: String
+    let isAdult: Bool
+    let overview: String
+    let releaseDate: String
+
+    enum Discover: String, CodingKey {
+        case identifier = "id"
+        case title
+        case votes = "vote_count"
+        case isVideo = "video"
+        case avarageVote = "vote_average"
+        case popularity
+        case poster = "poster_path"
+        case originalLanguage = "original_language"
+        case originalTitle = "original_title"
+        case genre = "genre_ids"
+        case backdrop = "backdrop_path"
+        case isAdult = "adult"
+        case overview
+        case releaseDate = "release_date"
+    }
+
+    public init(from decoder: Decoder) throws {
```

Comment:
why do you need to provide your own initializer instead of relying on `Decodable`'s auto-generated one?

Reply:
because I have custom `CodingKey`. I think this is necessary, isn't?

Reply:
Ok, now I get it 👍 
thanks guys